src/main/java/com/zufar/onlinestore/payment/endpoint/PaymentEndpoint.java

```diff
@@ -38,36 +37,63 @@ public class PaymentEndpoint {
+
+        ApiResponse<ProcessedPaymentWithClientSecretDto> apiResponse = ApiResponse.<ProcessedPaymentWithClientSecretDto>builder()
+                .data(processedPayment)
+                .message("Payment successfully processed")
+                .timestamp(LocalDateTime.now())
+                .httpStatusCode(HttpStatus.CREATED.value())
+                .build();
+
         return ResponseEntity.status(HttpStatus.CREATED)
-                .body(createdPayment);
+                .body(apiResponse);
     }
 
     @GetMapping("/{paymentId}")
-    public ResponseEntity<PaymentDetailsDto> getPaymentDetails(@PathVariable @NotNull final Long paymentId) throws PaymentNotFoundException {
-        PaymentDetailsDto retrievedPayment = paymentApi.getPaymentDetails(paymentId);
-        log.info("Get payment details: payment details: {} successfully retrieved.", retrievedPayment);
-        return ResponseEntity.ok()
-                .body(retrievedPayment);
-    }
+    public ResponseEntity<ApiResponse<ProcessedPaymentDetailsDto>> getPaymentDetails(@PathVariable @NotNull final Long paymentId) throws PaymentNotFoundException {
+        ProcessedPaymentDetailsDto retrievedPayment = paymentApi.getPaymentDetails(paymentId);
 
-    /**
-     * This endpoint is used only until we have an implementation of this logic on the frontend side.
-     * It will come in handy for testing the API.
-     */
-    @PostMapping("/method")
-    public ResponseEntity<String> getPaymentMethod(@RequestBody @Valid final CreatePaymentMethodDto createPaymentMethodDto) throws PaymentMethodProcessingException {
-        String paymentMethodId = paymentApi.createPaymentMethod(createPaymentMethodDto);
-        return ResponseEntity.ok()
-                .body(paymentMethodId);
+        ApiResponse<ProcessedPaymentDetailsDto> apiResponse = ApiResponse.<ProcessedPaymentDetailsDto>builder()
+                .data(retrievedPayment)
+                .message("Payment successfully retrieved")
+                .timestamp(LocalDateTime.now())
+                .httpStatusCode(HttpStatus.OK.value())
+                .build();
+
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(apiResponse);
     }
 
     @PostMapping("/event")
-    public ResponseEntity<Void> paymentEventsProcess(@RequestBody @NotEmpty final String paymentIntentPayload,
+    public ResponseEntity<ApiResponse<Void>> paymentEventProcess(@RequestBody @NotEmpty final String paymentIntentPayload,
                                                      @RequestHeader("Stripe-Signature") @NotEmpty final String stripeSignatureHeader) throws PaymentEventProcessingException, PaymentEventParsingException {
+
         paymentApi.processPaymentEvent(paymentIntentPayload, stripeSignatureHeader);
-        return ResponseEntity.ok()
+
+        ApiResponse<Void> apiResponse = ApiResponse.<Void>builder()
+                .message("Payment event successfully processed")
```

Comment:
probably needs to be in centralized storage

Reply:
Done